code/game/objects/items/weapons/melee/melee_misc.dm

```diff
@@ -45,6 +45,387 @@
+	update_icon()
+
+
+/obj/item/melee/secsword/attack(mob/M, mob/living/user)
+	if(HAS_TRAIT(user, TRAIT_CLUMSY) && prob(50))
+		if(state == SECSWORD_STUN && sword_stun(user, user, skip_cooldown = TRUE))
+			user.visible_message("<span class='danger'>[user] accidentally hits [user.p_themselves()] with [src]!</span>",
+							"<span class='userdanger'>You accidentally hit yourself with [src]!</span>")
+		return
+	if(user.mind?.martial_art?.no_baton && user.mind?.martial_art?.can_use(user)) // Just like the baton, no sword + judo.
+		to_chat(user, "<span class='warning'>The sword feels off-balance in your hand due to your specific martial training!</span>")
+		return
+
+	if(!isliving(M) || state == SECSWORD_OFF)
+		return ..()
+	if(state == SECSWORD_STUN) // Stamina
+		if(issilicon(M)) // Can't stun borgs and AIs
+			return ..()
+		sword_stun(M, user)
+		return ..()
+	//Burn
+	var/mob/living/L = M
+	if(ishuman(L))
+		var/mob/living/carbon/human/H = L
+		var/obj/item/organ/external/targetlimb = H.get_organ(ran_zone(user.zone_selected))
+		H.apply_damage(burn_damage, BURN, targetlimb, H.run_armor_check(targetlimb, MELEE))
+	else
+		L.apply_damage(burn_damage, BURN)
+	deduct_charge(burn_hitcost)
+	return ..()
+
+// Returning false results in no attack animation, returning true results in an animation.
+/obj/item/melee/secsword/proc/sword_stun(mob/living/L, mob/user, skip_cooldown = FALSE)
+	if(cooldown > world.time && !skip_cooldown)
+		return FALSE
+
+	var/user_UID = user.UID()
+	if(HAS_TRAIT_FROM(L, TRAIT_WAS_BATONNED, user_UID)) // Doesn't work in conjunction with stun batons.
+		return FALSE
+
+	cooldown = world.time + initial(cooldown) // Tracks the world.time when hitting will be next available.
+	if(ishuman(L))
+		var/mob/living/carbon/human/H = L
+		if(H.check_shields(src, 0, "[user]'s [name]", MELEE_ATTACK)) // No message; check_shields() handles that
+			playsound(L, 'sound/weapons/genhit.ogg', 50, TRUE)
+			return FALSE
+		// Weaker than a stun baton, less bad effects applied
+		H.Jitter(5 SECONDS)
+		var/obj/item/organ/external/targetlimb = H.get_organ(ran_zone(user.zone_selected))
+		H.apply_damage(stam_damage, STAMINA, targetlimb, H.run_armor_check(targetlimb, MELEE))
+		H.SetStuttering(5 SECONDS)
+		deduct_charge(stam_hitcost)
+
+	ADD_TRAIT(L, TRAIT_WAS_BATONNED, user_UID) // So a person cannot hit the same person with a sword AND a baton, or two swords
+	addtimer(CALLBACK(src, PROC_REF(stun_delay), L, user_UID), 2 SECONDS)
+	SEND_SIGNAL(L, COMSIG_LIVING_MINOR_SHOCK, 33)
+	playsound(src, 'sound/weapons/egloves.ogg', 50, TRUE, -1)
+	return TRUE
+
+//Proc called to remove trait that prevents repeated stamina damage. Called on a 2 Second timer when hit in stun mode
+/obj/item/melee/secsword/proc/stun_delay(mob/living/target, user_UID)
+	REMOVE_TRAIT(target, TRAIT_WAS_BATONNED, user_UID)
+
+/obj/item/melee/secsword/proc/deduct_charge(amount)
+	if(!cell)
+		return
+	cell.use(amount)
+	if(cell.rigged)
+		cell = null
```

Comment:
won't this just nullspace the cell inside instead of properly cleaning it up :/ ?